core/src/apps/cardano/sign_tx.py

```diff
@@ -586,51 +723,112 @@ async def _show_output(
     protocol_magic: int,
     network_id: int,
 ) -> None:
-    if output.address_parameters:
-        await _fail_or_warn_if_invalid_path(
-            ctx,
-            SCHEMA_ADDRESS,
-            output.address_parameters.address_n,
-            CHANGE_OUTPUT_PATH_NAME,
-        )
+    if signing_mode == CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER:
+        return
 
-        await _show_change_output_staking_warnings(
-            ctx, keychain, output.address_parameters, output.amount
-        )
+    if output.asset_groups_count > 0:
+        await show_warning_tx_output_contains_tokens(ctx)
```

Comment:
Why do we call 
```
        await show_warning_tx_output_contains_tokens(ctx)
```
even in case the output will be hidden, as determined later?

Reply:
It's a bug 🙈

Reply:
Since the function is only called when the output is being shown it's safe now to call it at the start of the function.